src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -146,6 +147,7 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 		this.add(this._viewDescriptorService.onDidChangeLocation(({ views }) => {
 			if (views.some(v => v.id === TERMINAL_VIEW_ID)) {
 				this._updateTheme();
+				this._decorationAddon?.refreshLayouts();
```

Comment:
this is necessary so when it moves from or to the panel, the correct left margin is used